### PR TITLE
Browse images from branch changes after v0.4

### PR DESCRIPTION
**Pull request by nemo794**
This PR adds the Browse Image feature to the DSWx-HLS code. It should be merged into the branch gshiroma/changes_after_v0.4.

Ideally, the Browse Image should have square dimensions to match the HLS granule dimensions. This is noted in the runconfig .yaml default file and .yaml schema file.

However, the output dimensions are flexible and can be user-specified via the runconfig. As an edge-case test, a test dataset was run with these non-standard dimensions as inputs in the runconfig.yaml:

```
browse_image_height: 345
browse_image_width: 654
```

This test image was output, with dimensions consistent with those non-standard test values:
OPERA_DSWx-HLS_L8_30_T11UQP_20210908T182436Z_20220823T224349Z_v0 5

Note that the size of the image in pixels matches the inputs that were specified.

### CODE REVIEW

File: src/proteus/dswx_hls.py [hunk 1]
```diff
@@ -2497,8 +2586,31 @@ def parse_runconfig_file(user_runconfig_file = None, args = None):
...
+        default_browse_fname = os.path.join(output_directory,
+                                            product_basename)
+
+        # If a browse image filename was provided via CLI, it takes
+        # precendence over the default filename.
+        if cli_browse_fname is not None:
+                logger.warning(f'command line {cli_arg_name} "{cli_browse_fname}" has'
+                                f' precedence over default {cli_arg_name}'
+                                f' "{default_browse_fname}".')
+                # `args` already contains the correct filename; no need to update.
```
Comment (gshiroma) [inline src/proteus/dswx_hls.py]:
```suggestion
            logger.warning(f'command line {cli_arg_name} "{cli_browse_fname}" has'
                            f' precedence over default {cli_arg_name}'
                            f' "{default_browse_fname}".')
            # `args` already contains the correct filename; no need to update.
```

File: src/proteus/dswx_hls.py [hunk 2]
```diff
@@ -1939,6 +1958,68 @@ def save_dswx_product(wtr, output_file, dswx_metadata_dict, geotransform,
     logger.info(f'file saved: {output_file}')
 
 
+def geotiff2png(src_geotiff_filename,
+                dest_png_filename,
+                output_height=-1,
+                output_width=-1,
```
Comment (gshiroma) [inline src/proteus/dswx_hls.py]: If you want to make these arguments optional, you probably need to handle them in case they are not provided by the user. In the rest of the code, you used the default value `None`. Is there a reason, why you are using `-1` here?

**Review by gshiroma — APPROVED** [review]
Looks great, @nemo794! Thank you so much for working on this!

File: src/proteus/dswx_hls.py (hunk 2 above)
Comment (gshiroma) [reply src/proteus/dswx_hls.py]: Just noticed that you fixed it in the last commit.

File: src/proteus/dswx_hls.py (hunk 1 above)
Comment (gshiroma) [reply src/proteus/dswx_hls.py]: The indentation of this block seems to be a bit messed up. Please check that in the text editor.

File: src/proteus/dswx_hls.py (hunk 2 above)
Comment (nemo794) [reply src/proteus/dswx_hls.py]: Oof, yeah. My apologies for that. I just went back to look at something, and realized I never handled those cases, and that `scratch_dir` never got used. (sigh) But really good catch! Thank you!

File: src/proteus/dswx_hls.py (hunk 1 above)
Comment (nemo794) [reply src/proteus/dswx_hls.py]: Fixed.
